packages/ptokens-entities/src/ptokens-asset-builder.ts

```diff
@@ -34,12 +32,12 @@ export abstract class pTokensAssetBuilder {
 
   /**
    * Set the blockchain chain ID for the token.
-   * @param _networkId - The chain ID.
+   * @param _chainId - The chain ID.
    * @returns The same builder. This allows methods chaining.
    */
-  setBlockchain(_networkId: NetworkId) {
-    if (networkIdToTypeMap.get(_networkId) !== this._type) throw new Error('Unsupported chain ID')
-    this._networkId = _networkId
+  setBlockchain(_chainId: ChainId) {
+    if (networkIdToTypeMap.get(_chainId) !== this._type) throw new Error('Unsupported chain ID')
```

Comment:
Following my previous comment: here I would use the AdapterAddress mapping instead to see if the chainId is supported or not